cloud/azure/deploy/website.go

```diff
@@ -0,0 +1,522 @@
+			},
+		},
+	}
+
+	ruleOrder := 1
+
+	// For each API forward to the appropriate API gateway
+	for name, resource := range p.Apis {
+		apiHostName := resource.ApiManagementService.GatewayUrl.ApplyT(func(gatewayUrl string) (string, error) {
+			parsed, err := url.Parse(gatewayUrl)
+			if err != nil {
+				return "", err
+			}
+
+			return parsed.Hostname(), nil
+		}).(pulumi.StringOutput)
+
+		apiOriginName := pulumi.Sprintf("api-origin-%s", name)
+
+		origins = append(origins, &cdn.DeepCreatedOriginArgs{
+			Name:             apiOriginName,
+			HostName:         apiHostName,
+			OriginHostHeader: apiHostName,
+			HttpPort:         pulumi.Int(80),
+			HttpsPort:        pulumi.Int(443),
+		})
+
+		apiOriginGroupName := fmt.Sprintf("api-origin-group-%s", name)
+
+		originGroups = append(originGroups, &cdn.DeepCreatedOriginGroupArgs{
+			Name: pulumi.String(apiOriginGroupName),
+			Origins: cdn.ResourceReferenceArray{
+				&cdn.ResourceReferenceArgs{
+					Id: p.getOriginId(profile.Name, endpointName, apiOriginName),
+				},
+			},
+		})
+
+		deliveryRuleOutput := pulumi.All(p.ResourceGroup.Name, profile.Name).ApplyT(func(args []interface{}) cdn.DeliveryRuleOutput {
+			resourceGroupName := args[0].(string)
+			profileName := args[1].(string)
+
+			apiOriginGroupId := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Cdn/profiles/%s/endpoints/%s/originGroups/%s",
+				p.ClientConfig.SubscriptionId,
+				resourceGroupName,
+				profileName,
+				endpointName,
+				apiOriginGroupName)
+
+			rule := &cdn.DeliveryRuleArgs{
+				Name:  pulumi.Sprintf("forward_%s", name),
+				Order: pulumi.Int(ruleOrder),
+				Conditions: pulumi.ToArray(
+					[]interface{}{
+						cdn.DeliveryRuleUrlPathCondition{
+							Name: "UrlPath",
+							Parameters: cdn.UrlPathMatchConditionParameters{
+								MatchValues: []string{
+									fmt.Sprintf("/api/%s", name),
+								},
+								Transforms: []string{
+									string(cdn.TransformLowercase),
+								},
+								TypeName: "DeliveryRuleUrlPathMatchConditionParameters",
+								Operator: string(cdn.OperatorBeginsWith),
+							},
+						},
+					}),
+				Actions: pulumi.ToArray(
+					[]interface{}{
+						cdn.OriginGroupOverrideAction{
+							Name: "OriginGroupOverride",
+							Parameters: cdn.OriginGroupOverrideActionParameters{
+								OriginGroup: cdn.ResourceReference{
+									Id: &apiOriginGroupId,
+								},
+								TypeName: "DeliveryRuleOriginGroupOverrideActionParameters",
+							},
+						},
+						cdn.UrlRewriteAction{
+							Name: "UrlRewrite",
+							Parameters: cdn.UrlRewriteActionParameters{
+								Destination:   "/",
+								SourcePattern: fmt.Sprintf("/api/%s/", name),
+								TypeName:      "DeliveryRuleUrlRewriteActionParameters",
+							},
+						},
+						// TODO add cache control
+					},
+				),
+			}
+
+			ruleOrder++
+
+			return rule.ToDeliveryRuleOutput()
+		}).(cdn.DeliveryRuleOutput)
+
+		deliveryRules = append(deliveryRules, deliveryRuleOutput)
+	}
```

Comment:
is order here always guaranteed between deployments if we're looping over map keys?

Reply:
changes made